Composer/packages/server/src/services/telemetry.ts

```diff
@@ -43,10 +57,16 @@ if (instrumentationKey) {
   AppInsights.defaultClient.addTelemetryProcessor((envelope: AppInsights.Contracts.Envelope): boolean => {
     const { sessionId, telemetry, composerVersion, userId, architecture, cpus, memory } = getTelemetryContext();
 
+    const data = envelope.data as AppInsights.Contracts.Data<AppInsights.Contracts.RequestData>;
+
+    if (alwaysTrackEvents.includes(data.baseData.name as TelemetryEventName) && !telemetry?.allowDataCollection) {
```

Comment:
We might run into a race condition here if the client hasn't updated the telemetry settings before the event gets logged. Maybe we make a requirement that all items in the alwaysTrackEvents array must include the `enabled` prop that we use to check if we should strip pii.